pkg/network/server.go

```diff
@@ -788,9 +813,13 @@ func verifyNotaryRequest(bc blockchainer.Blockchainer, _ *transaction.Transactio
 	if err := bc.VerifyWitness(payer, r, &r.Witness, bc.GetPolicer().GetMaxVerificationGAS()); err != nil {
 		return fmt.Errorf("bad P2PNotaryRequest payload witness: %w", err)
 	}
-	if r.FallbackTransaction.Sender() != bc.GetNotaryContractScriptHash() {
+	notaryHash := bc.GetNotaryContractScriptHash()
+	if r.FallbackTransaction.Sender() != notaryHash {
 		return errors.New("P2PNotary contract should be a sender of the fallback transaction")
 	}
+	if len(r.MainTransaction.Signers) != len(r.MainTransaction.Scripts) { // need this check: 1) to clearly define sig/multisig transaction type; 2) to preserve the order of witnesses in case of standard signature witnesses;
```

Comment:
Can we make this comment span 3 lines? It is hard to read.

Reply:
Fixed and also moved this check to NotaryRequestPayload's `isValid` because we don't need blockchain to perform this check.